Docs 3684 mmp v2 3 0

Cristian-Venticinque opened the pull request.

Commit:
Modified this page to show documentation for MMP version 2.3.x

Updated the mule-maven-plugin.adoc page to show the documentation for 
version 2.3.x of the Mule Maven Plugin. 
Information related to MMP 2.2.x was removed and included in a new page.

Commit:
Moved documentation for Mule Maven Plugin 2.2.x to a new page.

Created a new page called mule-maven-plugin-v2.2.adoc keeping all 
information of this version of the MMP.

Commit:
Separated Mule Maven Plugin into 2 different pages (2.2.x and 2.3.x)

MMP has different usage and implementation instructions depending of the 
version so a new article was created for MMP 2.3.x

Commit:
Added a new line to properly show a [Note] element.

julianpascualMuleSoft commented on modules/ROOT/pages/mule-maven-plugin-v2.2.adoc:
```diff
@@ -0,0 +1,789 @@
+
+=== Adding the Maven Dependency for the Plugin
+
+Edit your `settings.xml` or project file to include the following:
```
Should it be pom.xml file instead.

julianpascualMuleSoft commented on modules/ROOT/pages/mule-maven-plugin-v2.2.adoc:
```diff
@@ -0,0 +1,789 @@
+
+=== Adding the Maven Repository
+
+The repository for the plugin is at https://repository.mulesoft.org/nexus/content/repositories/releases. To add it to your Maven installation, edit your `settings.xml` or project file to include the following:
```
Same as previous comment, this config should go on pom.xml

julianpascualMuleSoft commented on modules/ROOT/pages/mule-maven-plugin.adoc:
```diff
@@ -56,730 +39,581 @@ The repository for the plugin is at https://repository.mulesoft.org/nexus/conten
+CloudHub rewrites the app you had deployed.
 
-To run integration tests, the basic steps are the following:
+=== Encrypting credentials
```
Could you move this section at the bottom of this doc? Because the server parameter to pass encrypted credentials is available on all the platform deployments (Cloudhub, ARM, Runtime Fabric)

Commit:
Updated references to pom.xml file. 

The article referenced the settings.xml file for the MMP configurations 
but this should be in the pom.xml file.

Commit:
Moved the 'Encrypting Credentials' section out of CloudHub.

This section is common to CloudHub, Runtime Fabric and ARM so it was 
moved outside.
Added 'server' parameter description to Runtime Fabric and ARM 
references. 
Minor additional corrections.

Commit:
Adding a missing [NOTE] element in 'Encrypting Credentials' section.
